test/links_test.rb

```diff
@@ -295,6 +295,16 @@
     assert_xpath '//a[@href="tigers.html"][text() = "[tigers]"]', doc.render, 1
   end
 
+  test 'inter-document xref should not truncate after period if path has no extension' do
+    result = render_embedded_string '<<using-.net-web-services#,Using .NET web services>>'
+    assert_xpath '//a[@href="using-.net-web-services.html"][text() = "Using .NET web services"]', result, 1
+  end
+
+  test 'inter-document xref should only truncate file extension if path contains period' do
```

Comment:
"if path contains a known AsciiDoc extensions" ?
In my opinion, the title of this test is a bit misleading since the path contains two periods `using-.net-web-services.adoc` (ie. `.net` and `.adoc`)